### ISSUE TITLE
Current functionExit can't distinguish between a normal functionExit and a function exit with "undefined" exception

### ISSUE THREAD

**Issue by Haiyang-Sun** [Haiyang-Sun/nodeprof.js#17]
Regarding ```this.functionExit(iid, returnVal, exceptionVal)```:

Currently, NodeProf passes exception thrown in a function as it is to the callback.
In this case, a function throws an ```undefined``` value would have the same event as a function exits normally and returns ```undefined```

To fix this, we need to wrap the exceptionVal into an object as Jalangi does.

**Comment by eleinadani** [comment]
perhaps you can consider using the `onReturnExceptional` instrumentation trap in Graal.js to track exceptions?

**Comment by Haiyang-Sun** [comment]
@eleinadani it's not about this. We have used this trop already to generate the exception event. It's just about a flaw in NodeProf's current callback design for ```functionExit```.

Currently, the following two cases will have the same events:
```
function foo(){
    throw undefined;
}
foo();
```

```
function foo(){
    return undefined;
}
foo();
```

And this is just because NodeProf pass directly the exception caught to the callback. We can fix this by wrapping the exception thrown into a TruffleObject, and we would know that if the function returns normally or exits with "undefined" exception.

**Comment by Haiyang-Sun** [comment]
This issue is used to remember the problem, and I will fix it once I have some time.

**Comment by eleinadani** [comment]
OK, thanks for explaining. In this case:
```
function foo() {
    throw undefined;
}
```
you would still detect the exception event that you would not see with the normal `return`, right?

**Comment by Haiyang-Sun** [comment]
Yes, we can detect that.

**Comment by alexjordan** [comment]
Good point. I totally overlooked that the API differs from the original Jalangi here. So I guess the plan would be, to solve it in the same way:
```
this.functionExit = function (iid, returnVal, wrappedExceptionVal) {}
```
where `wrappedExceptionVal` is `undefined` if there was no exception and otherwise it's an object like this: `{ exception: <theExceptionValue> }`.

**Comment by alexjordan** [comment]
Fixed in #52
